repositories/summary.go

```diff
@@ -36,28 +36,42 @@ func (r *SummaryRepository) GetAll() ([]*models.Summary, error) {
 
 func (r *SummaryRepository) Insert(summary *models.Summary) error {
 
-	itemsToCreate := []interface{}{summary}
+	if err := r.db.Transaction(func(tx *gorm.DB) error {
+		if err := r.db.Create(summary).Error; err != nil {
```

Comment:
Doesn't seem to be working for me, sorry. When letting the `AggregationService` create summaries on startup, it would get stuck while trying to persist the very first summary in list. Precisely, it seems to get stuck it this very line, that is, the `Create()` would never return. Probably has got to do with the transaction management. I'm using MySQL. 

These are the logs:

```
2024-02-02T14:58:00.019589431+01:00 [INFO ] generating summaries
2024/02/02 14:58:00 /home/ferdinand/go/src/github.com/muety/wakapi/repositories/summary.go:114
[55.661ms] [rows:270] SELECT users.id as `user`, max(to_time) as time FROM `users` left join summaries on users.id = summaries.user_id GROUP BY `users`.`id`
2024/02/02 14:58:07 /home/ferdinand/go/src/github.com/muety/wakapi/repositories/heartbeat.go:140
[7594.383ms] [rows:270] SELECT users.id as `user`, min(time) as `time` FROM `users` left join heartbeats on users.id = heartbeats.user_id GROUP BY `users`.`id`
2024/02/02 14:58:10 /home/ferdinand/go/src/github.com/muety/wakapi/repositories/heartbeat.go:97
[48.463ms] [rows:0] SELECT * FROM `heartbeats` WHERE `heartbeats`.`user_id` = 'n1try' AND time >= '2021-02-11 00:00:00' AND time < '2021-02-12 00:00:00' ORDER BY time asc
2024/02/02 14:58:10 /home/ferdinand/go/src/github.com/muety/wakapi/repositories/language_mapping.go:42
[0.726ms] [rows:2] SELECT * FROM `language_mappings` WHERE `language_mappings`.`user_id` = 'n1try'
2024-02-02T14:58:10.921284153+01:00 [INFO ] successfully generated summary (2021-02-11 00:00:00 +0100 CET, 2021-02-12 00:00:00 +0100 CET, n1try)
```

Sorry for the struggle! Let me know if you want me to try to take over.

Reply:
Sorry my bad. I should use `tx` instead of the original `r.db` to create the entities. I rewrote the code because the master has changed since the original fix, and just missed the part. I just pushed a new commit to resolve this, please check the latest one.